Move Cuda Abs kernel to its own file.

gchanan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #38275 Don't double dispatch on CUDA logical_not.
* **#38274 Move Cuda Abs kernel to its own file.**

UnarySignKernels is one of the longest files to compile and Abs is not a sign function.

Differential Revision: [D21511831](https://our.internmc.facebook.com/intern/diff/D21511831)

gchanan committed:
Move Cuda Abs kernel to its own file.

UnarySignKernels is one of the longest files to compile and Abs is not a sign function.

[ghstack-poisoned]

mruberry commented on aten/src/ATen/native/cuda/AbsKernel.cu:
```diff
@@ -0,0 +1,43 @@
+#include <ATen/Dispatch.h>
+#include <ATen/native/DispatchStub.h>
+#include <ATen/native/TensorIterator.h>
+//#include <ATen/native/cuda/Math.cuh>
```
Commented out?

gchanan replied:
ah ya, I was checking that it wasn't needed, I'll just remove it.

mruberry reviewed (approved):
Stamped.

gchanan committed:
Update on "Move Cuda Abs kernel to its own file."


UnarySignKernels is one of the longest files to compile and Abs is not a sign function.

Differential Revision: [D21511831](https://our.internmc.facebook.com/intern/diff/D21511831)

[ghstack-poisoned]

gchanan committed:
Update on "Move Cuda Abs kernel to its own file."


UnarySignKernels is one of the longest files to compile and Abs is not a sign function.

Differential Revision: [D21511831](https://our.internmc.facebook.com/intern/diff/D21511831)

[ghstack-poisoned]

gchanan committed:
Update on "Move Cuda Abs kernel to its own file."


UnarySignKernels is one of the longest files to compile and Abs is not a sign function.

Differential Revision: [D21511831](https://our.internmc.facebook.com/intern/diff/D21511831)

[ghstack-poisoned]

gchanan committed:
Update on "Move Cuda Abs kernel to its own file."


UnarySignKernels is one of the longest files to compile and Abs is not a sign function.

Differential Revision: [D21511831](https://our.internmc.facebook.com/intern/diff/D21511831)

[ghstack-poisoned]

jeffdaily commented:
@gchanan please check #38380 for the correct/updated implementation of `aten/src/ATen/native/cuda/AbsKernel.cu`.  CC @zasdfgbnm

gchanan committed:
Update on "Move Cuda Abs kernel to its own file."


UnarySignKernels is one of the longest files to compile and Abs is not a sign function.

Differential Revision: [D21511831](https://our.internmc.facebook.com/intern/diff/D21511831)

[ghstack-poisoned]

zasdfgbnm commented on aten/src/ATen/native/cuda/AbsKernel.cu:
```diff
@@ -0,0 +1,40 @@
+
+namespace at { namespace native {
+
+// We manually overload abs because std::abs does not work with thrust::complex types and ROCm.
```
Nit: I forgot to update this in my original PR. Should be:
```
// We manually overload abs because std::abs does not work with ROCm.
```
We are no longer using `thrust::complex` any more

gchanan commented:
@zasdfgbnm thanks, I'll put that in a follow up PR.
